### ISSUE TITLE
Organize and stabilize the stuff

### ISSUE THREAD

**Issue by Wohlstand** [Wohlstand/TheXTech#172]
Once we started to work on TheXTech together and we had discussed new features and biggies, there is still the old work that was done previously before the agreement on organizing features was done. So, as the result the `master` branch doesn't look like a stable thing: while it has several major improvements and new things, it's unstable and has some serious bugs (first off, missing blocks render bug and #171) making it unsuitable for quick releases yet. Also, it has several unpolished code parts that need future work and review from my side.

# The plan
- [x] Rename branches: `master` -> `devel`, and `stable-1.3.5.x` -> `master`. So, the goal of the `master` branch is to keep the latest stable state that is ready for quick releases at any moment (also allowing people to get the freshest but almost stable build from the [builds hive](https://github.com/Wohlstand/TheXTech/wiki/Fresh-devel-builds) directly). The `devel` branch will be intended to be a playground for new features and other things to be developed safely. And other feature branches intended for various biggies.
- [x] Proceed with the task lists for the 1.3.5.2 version and release it
- ~~Choose the best (in a stability term) features at the `devel` branch, and gradually backport each of them over the `master` (such as quad-tree support for blocks, and implement the quad-tree for other object types, backport the improved moving layers, and, backport the multi-resolution system support)~~ Moved to #232
- [x] Develop every new feature over the latest stable state directly to guarantee the general stability and avoid messing up the mixture of new features being not properly polished

So, the development of all new features should go consistently and no mixing up between biggies should be done, guarantee all code can be reviewed by us all and managed properly.

I made this issue to make sure to avoid any sudden surprises resulting from renamed branches, and I need confirmation from all of us that we are fine with this.

P.S. Also, let's review the code of each other (also mine) to make sure no mistakes were done, and also, suggest better ideas if something looks possible to be improved in future

**Comment by Wohlstand** [comment]
Branches have been renamed, so, to avoid conflicts, please run these commands:
```
git branch -m master devel
git fetch origin
git branch -u origin/devel devel
git remote set-head origin -a

git branch -m stable-1.3.5.x master
git fetch origin
git branch -u origin/master master
git remote set-head origin -a
```
otherwise, you'll get the strange merge conflict on your end.

**Comment by LoveBodhi** [comment]
Will include parallax (Multi-layer) BGs for the new feature?

**Comment by Wohlstand** [comment]
There is another plan to backport Moondust Engine's background engine to support that here, but it wasn't made yet

**Comment by Wohlstand** [comment]
Closed in favor of #232
